elytron/src/main/java/org/wildfly/extension/elytron/TrivialAddHandler.java

```diff
@@ -46,16 +48,22 @@
 
     private final RuntimeCapability<?> runtimeCapability;
     private final Mode initialMode;
+    private final Mode embeddedInitialMode;
 
     TrivialAddHandler(Class<T> serviceType, AttributeDefinition[] attributes, RuntimeCapability<?> runtimeCapability) {
         this(serviceType, Mode.ACTIVE, attributes, runtimeCapability);
     }
 
     TrivialAddHandler(Class<T> serviceType, Mode initialMode, AttributeDefinition[] attributes, RuntimeCapability<?> runtimeCapability) {
+        this(serviceType, initialMode, initialMode, attributes, runtimeCapability);
+    }
+
+    TrivialAddHandler(Class<T> serviceType, Mode initialMode, Mode embeddedInitialMode, AttributeDefinition[] attributes, RuntimeCapability<?> runtimeCapability) {
         super(new HashSet<>(Collections.singletonList(checkNotNullParam("runtimeCapabilities", runtimeCapability))), attributes);
         this.runtimeCapability = runtimeCapability;
         checkNotNullParam("serviceType", serviceType);
         this.initialMode = checkNotNullParam("initialMode", initialMode);
+        this.embeddedInitialMode = checkNotNullParam("initialMode", embeddedInitialMode);
```

Comment:
This is a mistake, param must be "embeddedInitialMode" I'll fix it after getting the CI results.

Reply:
Fixed